src/SymbolTable.ts

```diff
@@ -65,15 +89,15 @@ export class SymbolTable {
         if (symbols?.length > 1) {
             //Check if each time it was set, it was set to the same type
             // TODO handle union types
-            let sameType = true;
-            const firstType = symbols[0].type.toString();
+            let sameImpliedType = true;
+            let impliedType = symbols[0].type;
             for (const symbol of symbols) {
-                sameType = firstType === symbol.type.toString();
-                if (!sameType) {
+                sameImpliedType = (impliedType.toString() === symbol.type.toString());
```

Comment:
You don't have to address this right now, but we should probably have a `.equals()` method on each `Type` so we can compare them more precisely.